src/server/generic_family.cc

```diff
@@ -734,6 +734,43 @@ void GenericFamily::Persist(CmdArgList args, ConnectionContext* cntx) {
     (*cntx)->SendLong(0);
 }
 
+std::optional<int32_t> parseExpireOptions(const CmdArgList args, ConnectionContext* cntx) {
+  int32_t flags = ExpireFlags::EXPIRE_ALWAYS;
+  bool nx{};
+  bool xx{};
+  bool gt{};
+  bool lt{};
+  for (auto& arg : args) {
+    ToUpper(&arg);
+    auto myArg = ToSV(arg);
+    if (myArg == "NX") {
+      flags |= ExpireFlags::EXPIRE_NX;
+      nx = true;
+    } else if (myArg == "XX") {
+      flags |= ExpireFlags::EXPIRE_XX;
+      xx = true;
+    } else if (myArg == "GT") {
+      flags |= ExpireFlags::EXPIRE_GT;
+      gt = true;
+    } else if (myArg == "LT") {
+      flags |= ExpireFlags::EXPIRE_LT;
+      lt = true;
+    } else {
+      (*cntx)->SendError(absl::StrCat("Unsupported option: ", ToSV(arg)));
+      return nullopt;
+    }
+  }
+  if ((nx && xx) || (nx && gt) || (nx && lt)) {
+    (*cntx)->SendError("NX and XX, GT or LT options at the same time are not compatible");
+    return nullopt;
```

Comment:
you can check `(flags & NX) && (flags & ~NX)`, which means that both NX and some other option are set, its quite compact, so you don't need the bool variables